UWP/PPSSPP_UWPMain.cpp

```diff
@@ -462,48 +501,95 @@ void System_Notify(SystemNotification notification) {
 
 bool System_MakeRequest(SystemRequestType type, int requestId, const std::string &param1, const std::string &param2, int param3) {
 	switch (type) {
+
+	case SystemRequestType::EXIT_APP:
+	{
+		bool state = false;
+		ExecuteTask(state, Windows::UI::ViewManagement::ApplicationView::GetForCurrentView()->TryConsolidateAsync());
+		if (!state) {
+			// Notify the user?
+		}
+		return true;
+	}
+	case SystemRequestType::RESTART_APP:
+	{
+		Windows::ApplicationModel::Core::AppRestartFailureReason error;
+		ExecuteTask(error, Windows::ApplicationModel::Core::CoreApplication::RequestRestartAsync(nullptr));
+		if (error != Windows::ApplicationModel::Core::AppRestartFailureReason::RestartPending) {
+			// Shutdown
+			System_MakeRequest(SystemRequestType::EXIT_APP, requestId, param1, param2, param3);
+		}
+		return true;
+	}
 	case SystemRequestType::BROWSE_FOR_IMAGE:
-	case SystemRequestType::BROWSE_FOR_FILE:
 	{
-		auto picker = ref new Windows::Storage::Pickers::FileOpenPicker();
-		picker->ViewMode = Pickers::PickerViewMode::List;
+		std::vector<std::string> supportedExtensions = { ".jpg", ".png" };
 
-		if (type == SystemRequestType::BROWSE_FOR_IMAGE) {
-			picker->FileTypeFilter->Append(".jpg");
-			picker->FileTypeFilter->Append(".png");
-		} else {
-			switch ((BrowseFileType)param3) {
-			case BrowseFileType::BOOTABLE:
-				// These are single files that can be loaded directly using StorageFileLoader.
-				picker->FileTypeFilter->Append(".cso");
-				picker->FileTypeFilter->Append(".iso");
-
-				// Can't load these this way currently, they require mounting the underlying folder.
-				picker->FileTypeFilter->Append(".bin");
-				picker->FileTypeFilter->Append(".elf");
-				break;
-			case BrowseFileType::INI:
-				picker->FileTypeFilter->Append(".ini");
-				break;
-			case BrowseFileType::DB:
-				picker->FileTypeFilter->Append(".db");
-				break;
-			case BrowseFileType::ANY:
-				picker->FileTypeFilter->Append("*");
-				break;
+		//Call file picker
+		ChooseFile(supportedExtensions).then([requestId](std::string filePath) {
+			if (filePath.size() > 1) {
+				g_requestManager.PostSystemSuccess(requestId, filePath.c_str());
 			}
+			else {
+				g_requestManager.PostSystemFailure(requestId);
+			}
+			});
+		return true;
+	}
+	case SystemRequestType::BROWSE_FOR_FILE:
+	{
+		std::vector<std::string> supportedExtensions = { ".cso", ".bin", ".iso", ".elf", ".zip" };
+		switch ((BrowseFileType)param3) {
+		case BrowseFileType::INI:
+			supportedExtensions = { ".ini" };
+			break;
+		case BrowseFileType::DB:
+			supportedExtensions = { ".db" };
+			break;
+		case BrowseFileType::ANY:
+			supportedExtensions = {};
+			break;
```

Comment:
I recommend using a `default:` here.  For defensive programming, the best would be:

```
switch ((BrowseFileType)param3) {
...
case BrowseFileType::BOOTABLE:
	break;
default:
	ERROR_LOG(FILESYS, "Unexpected BrowseFileType: %d", param3);
	return false;
}
```

This is called "defensive programming."  It's a technique where you assume other parts of the code will change, and the person who changes them will not realize your code also needs to change.  By logging and returning false, it becomes more likely someone will figure out the cause of the error instead of making them spend hours debugging why something doesn't work, not realizing it's because the case is missing here.

It's almost a certainty that this will eventually change in PPSSPP and the person who changes it will very likely forget to update UWP.

-[Unknown]

Reply:
Hm, wait, you didn't add `BrowseFileType::BOOTABLE`, so I think this will fail for that type of dialog currently.

-[Unknown]

Reply:
Oops, I shouldn't forget to add it back sorry about that